### PR TITLE
Add Grunion Editor View.

### PR DESCRIPTION
**Pull request by georgestephanis**
Related: #2065

contact-form

This is currently just adding the overrides -- I'm intentionally not removing the old legacy code yet, to ease the transition on any users that prefer the old method.  They can just disable the new include line, and the old way will be back again.

Props @MichaelArestad @azaozz and @georgestephanis for building this at the 2016 a8c GM.

#### Changes proposed in this Pull Request:

* Add a visual interface for Contact Forms.

#### Testing instructions:

* Add a contact form to a post.
* Edit an existing contact form in a post.
* Switch to the html editor.  Back again!
* Add a second or third contact form to a post.
* Open multiple forms to editing mode at the same time.

#### Proposed changelog entry for your changes:

* Added a new inline preview and editing interface for Contact Forms.

### CODE REVIEW

**Comment by georgestephanis** [comment]
Probably will need RTL and min assets generated.

**Comment by georgestephanis** [comment]
We're also moving the "Add Contact Form" button directly into the TinyMCE Toolbar.

**Comment by georgestephanis** [comment]
This was developed originally at https://github.com/Automattic/gm2016-grunion-editor -- so it has the full history there.

**Comment by MichaelArestad** [comment]
I would like to dramatically overhaul the styles here. I'm not sure that's doable in time for this release considering all the things we're wrangling in settings land. Can we plan for this to be in the following release to give us a little time to polish it?

**Comment by georgestephanis** [comment]
@michaelarestad There is no target for merge, so no rush.

**Comment by jeherve** [comment]
Fixes #1849
Fixes #1055
Fixes #642
Fixes #604
Fixes #254

Once this is merged, we should also retest the following issues and see if it helps:
#3109
#663

**Comment by jeherve** [comment]
Fixes #7063

**Comment by georgestephanis** [comment]
@michaelarestad Do you feasibly think you'll be working on this or should we ship and iterate?

**Comment by georgestephanis** [comment]
The diff to implement this on wpcom is at D6457

File: modules/contact-form/grunion-editor-view.php [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+		wp_localize_script( 'grunion-editor-view', 'grunionEditorView', array(
+			'inline_editing_style' => plugins_url( 'css/editor-inline-editing-style.css', __FILE__ ),
+			'dashicons_css_url'    => includes_url( 'css/dashicons.css' ),
+			'default_form'  => '[contact-field label="' . __( 'Name', 'jetpack' ) . '" type="name"  required="true" /]' .
+								'[contact-field label="' . __( 'Email', 'jetpack' )   . '" type="email" required="true" /]' .
+								'[contact-field label="' . __( 'Website', 'jetpack' ) . '" type="url" /]' .
+								'[contact-field label="' . __( 'Message', 'jetpack' ) . '" type="textarea" /]',
+			'labels'      => array(
+				'submit_button_text'  => __( 'Submit', 'jetpack' ),
+				'required_field_text' => apply_filters( 'jetpack_required_field_text', __( '(required)', 'jetpack' ) ),
```
Comment (eliorivero) [inline modules/contact-form/grunion-editor-view.php]: This needs a phpdoc so it's collected by our inline doc parser.

**Comment by georgestephanis** [comment]
The prior mentioned phabricator diff has been updated with the latest changes.

**Comment by georgestephanis** [comment]
Well, rebasing made this messier than expected.

**Pull request closed without merging**